### PR TITLE
implement NpcEventAPI

### PR DESCRIPTION
**Pull request by anjeshshrestha**
API interface for NpcEvent proxy

### CODE REVIEW

File: api/src/main/java/eu/darkbot/api/managers/NpcEventAPI.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * Provides data for any Npc event
+ */
+public interface NpcEventAPI extends API.Singleton {
+
+    /**
+     * Get how long until the current {@link #getStatus()} changes.
+     * @return how long until the npc is active/inactive, in seconds
+     */
+    double getRemainingTime();
+    /**
```
Comment (Pablete1234) [inline api/src/main/java/eu/darkbot/api/managers/NpcEventAPI.java:15]:
```suggestion

    /**
```
Spacing between methods

**Review by Pablete1234 — COMMENTED** [review]
Looks incomplete, i assume you're still trying to define this API on the impl side to decide what should be exposed?

**Comment by anjeshshrestha** [comment]
I think its sort of done, on the case that its possible to detect event has ended.

The extras about npc event name, description, npc left. This im not sure how it will different from event to event, I can also expose these?

**Comment by Pablete1234** [comment]
You should try to expose anything that plugins may have a reason to use, this api as-is seems needlessly restrictive, in the sense that it provides extremely little info

**Review by Pablete1234 — APPROVED** [review]

**Pull request merged**